src/lib/migration/repository_checker.rb

```diff
@@ -37,6 +37,9 @@ def initialize(products)
     def obsolete_product_repos
       old_repos = obsolete_available_products.map { |product| product["source"] }
 
+      # make sure the system repo or invalid values are filtered out
```

Comment:
NTH: Maybe a reference to the bug resulting in the -1s here, for future reference?